src/server/services/leaderboard.service.ts

```diff
@@ -179,6 +181,8 @@ export async function getLeaderboard(input: GetLeaderboardInput) {
   );
 }
 
+export type LeaderboardWithResults = Awaited<ReturnType<typeof getLeaderboardsWithResults>>[number];
+
```

Comment:
It'd probably be a good idea to put this in `~/types/router` and get it from the `leaderboard.router` endpoint